Convert CSV to binary (#19)
CUREd-Plus/cuRed: CONTRIBUTING.md

|:-------------------------|:--------------------------------------|:--------------------------------------------------|:-----------------------------------------------------------------------------------|
| `ns-rse/1-package-setup` | [`ns-rse`](https://github.com/ns-rse) | [1](https://github.com/CUREd-Plus/cuRed/issues/1) | `package-setup` short for the issue subject *Package setup, checking and linting*. |

## Coding Standards

To make the codebase easier to maintain we ask that you follow the
guidelines below on coding style, linting, typing, documentation and
testing.

### Coding Style/Linting

Using a consistent coding style has many benefits (see [Linting : What
is all the fluff
about?](https://rse.shef.ac.uk/blog/2022-04-19-linting/)). For this
project we aim to adhere to [`lintr`](https://github.com/r-lib/lintr/)
styles and for the package structure we are following the [R Packages
(2e)](https://r-pkgs.org/) guidelines.

`pre-commit` hooks are used to ensure these are consistent before Git
commits are made.

#### `pre-commit`

[pre-commit](https://pre-commit.com/) runs Git
[hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) that

> it is configured to run on the code base prior to Git commits being

Reviewer: Minor

```suggestion
are configured to run on the code base prior to Git commits being
```

made. It is highly configurable but as it is written in Python requires
installing in order to analyse the R code and package structure of
`cuRed`. If you are not already familiar with Python and the various
options for Virtual Environments the simplest solution will likely be to
install [miniconda](https://docs.conda.io/en/latest/miniconda.html) and
then install the `pre-commit` package and install within the repository

``` bash
conda install conda-forge pre-commit
cd path/to/cloned/cuRed
pre-commit install --install-hooks
```

> 

Reviewer: I don't know if its worth mentioning here that...
```suggestion
On your first commit `pre-commit` will download a virtual environment from the R [precommit package](https://github.com/lorenzwalthert/precommit) which is used to run all the configured tests. You may find the [documentation](https://lorenzwalthert.github.io/precommit/) a useful references as well as the GitHub repositories issues.
```

For more an overview of `pre-commit` see the post [pre-commit:
Protecting your future self](https://rse.shef.ac.uk/blog/pre-commit/).

## Line endings

> To automatically use LF line endings in RStudio, open the following menu: Tools → Code → Saving

Reviewer: Should we explain what this is?
```suggestion
Windows uses a different method of starting new lines known as **CRLF** 
(**C**arriage **R**eturn **L**ine **F**eed), whilst other operating
systems (GNU/Linux, UNIX, OSX) use just **LF**. It saves some problems 
to be consistent in usage across development environments. Fortunately RStudio can be configured to use just **LF** when editing files.

To automatically use LF line endings in RStudio, open the following menu: Tools → Code → Saving
```

* Under "General" select "Ensure that source files end with newline"
* Under "Serialization" set the Line ending conversion to "Posix (LF)"

Git can also help with this. See: [Configuring Git to handle line endings](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings).

# Writing R packages

Please refer to this documentation for authoring R packages:

* R manual [Writing R Extensions](https://cran.r-project.org/doc/manuals/r-release/R-exts.html)
* [R Packages (2e)](https://r-pkgs.org/) by Hadley Wickham and Jennifer Bryan.

# README.md

> To automatically generate `README.md` from the `README.Rmd` file, [use this](https://usethis.r-lib.org/) to [create README files](https://usethis.r-lib.org/reference/use_readme_rmd.html) to automatically configure pre-commit hook that helps keep these files synchronized.

Reviewer: ```suggestion
The `README.md` is automatically generated from the `README.Rmd` file, courtesy of 
an additional pre-commit hook that is _not_ included in the `pre-commit` 
configuration. In order to enable this hook locally you must use the  
[usethis](https://usethis.r-lib.org/) function 
[`user_redme_rmd()`](https://usethis.r-lib.org/reference/use_readme_rmd.html) to install the 
pre-commit hook  (it is added to the `.git/hooks/pre-commit` file). Once run updates to
the `README.Rmd` will, on being committed, trigger the regeneration of the `README.md`.
```
